src/pages/abs/[id]/metrics.tsx

```diff
@@ -1,8 +1,139 @@
-import { NextPage } from 'next';
+import AdsApi, { IADSApiMetricsParams, IDocsEntity, IUserData } from '@api';
+import { MetricsResponseKey, CitationsStatsKey, BasicStatsKey } from '@api/lib/metrics/types';
+import { AbsLayout } from '@components/Layout/AbsLayout';
+import { Metrics } from '@components/Metrics';
+import {
+  plotCitationsHist,
+  plotReadsHist,
+  getCitationTableData,
+  getReadsTableData,
+} from '@components/Metrics/graphUtils';
+import { ICitationsGraphData, IReadsGraphData, ICitationsTableData, IReadsTableData } from '@components/Metrics/types';
+import { normalizeURLParams, getDocument, getHasGraphics, getHasMetrics } from '@utils';
+import { NextPage, GetServerSideProps } from 'next';
 import React from 'react';
 
-const BasicPage: NextPage = () => {
-  return <div>basic page</div>;
+interface IMetricsPageProps {
+  originalDoc: IDocsEntity;
+  error?: string;
+  hasGraphics: boolean;
+  hasMetrics: boolean;
+  citationsGraph: ICitationsGraphData;
+  readsGraph: IReadsGraphData;
+  citationsTable: ICitationsTableData;
+  readsTable: IReadsTableData;
+}
+
+const MetricsPage: NextPage<IMetricsPageProps> = (props: IMetricsPageProps) => {
+  const { originalDoc, error, hasGraphics, hasMetrics, citationsGraph, readsGraph, citationsTable, readsTable } = props;
+
+  return (
+    <AbsLayout doc={originalDoc} hasGraphics={hasGraphics} hasMetrics={hasMetrics}>
+      <article aria-labelledby="title" className="mx-0 my-10 px-4 w-full bg-white md:mx-2">
+        <div className="pb-1">
+          <h2 className="prose-xl pb-5 text-gray-900 text-2xl font-medium leading-8" id="title">
+            <span>Metrics for </span> <div className="text-2xl">{originalDoc.title}</div>
+          </h2>
+        </div>
+        {error ? (
+          <div className="flex items-center justify-center w-full h-full text-xl">{error}</div>
+        ) : (
+          <Metrics
+            citationsGraph={citationsGraph}
+            readsGraph={readsGraph}
+            citationsTable={citationsTable}
+            readsTable={readsTable}
+            isAbstract={true}
+          />
+        )}
+      </article>
+    </AbsLayout>
+  );
 };
 
-export default BasicPage;
+export default MetricsPage;
+
+export const getServerSideProps: GetServerSideProps<IMetricsPageProps> = async (ctx) => {
```

Comment:
So I think we may want to try to simplify the serversideProps part here.  
We really only need the bibcode available on the page, then pass that as a prop to the `<Metrics>` component.  From there we should be doing all this extra stuff, that makes sure we don't have these interdependencies.

So I would say let's move all the metrics calls here to inside the component, we could make a custom hook or whatever, but that way we keep things encapsulated a little better

Reply:
The server is going to render the `<Metrics />` component, so it'll make all these calls anyway, but we can keep the page simpler